src/plugins/ciu/ciuPlugin.js

```diff
@@ -0,0 +1,71 @@
+const db = require('caniuse-db/data.json');
+const { messageToFactoid } = require('../factoids/factoidsPlugin');
+
+const features = Object.keys(db.data);
+
+const AGENTS_TO_SHOW = {
+  ie: 'IE',
+  edge: 'Edge',
+  firefox: 'FF',
+  chrome: 'Chrome',
+  opera: 'Opera',
+  safari: 'Safari',
+  ios_saf: 'iOS',
+  android: 'Android',
+};
+
+function getVersions(feature) {
+  return Object.entries(AGENTS_TO_SHOW).flatMap(([agent, agentName]) => {
+    // agents versions strings are sorted
+    const versionFull = db.agents[agent].versions.find(
+      (v) => v && !/[nupac]/g.test(feature.stats[agent][v]),
+    );
+
+    if (versionFull) return [`${agentName}${versionFull}`];
+
+    const versionPartial = db.agents[agent].versions.find(
+      (v) => v && !/[nu]/g.test(feature.stats[agent][v]),
+    );
```

Comment:
```suggestion
    const versions = db.agents[agent].versions.filter(Boolean);
    const versionFull = versions.find((v) => !/[nupac]/g.test(feature.stats[agent][v]));

    if (versionFull) return [`${agentName}${versionFull}`];

    const versionPartial = versions.find((v) => !/[nu]/g.test(feature.stats[agent][v]));
```

Reply:
not convinced :), cost of adding an intermediate variables vs cost of cheching just twice for `v` not `null`, for me second option is still fine

Reply:
it's not about cost, it's about clarity, which more intermediate variables improves.